### PR TITLE
Chown - Check for dangerous root operations through folders and symlinks

### LINKED ISSUE #1709: chown does not check for symlinks pointing to root when recursing through a tree
Even if the `--preserve-root` option is set the chown command will attempt to follow a symlink pointing to root. This can be seen with the following test:
```
mkdir point-to-root
ln -s / point-to-root/root
chown -RHh --preserve-root 1000 point-to-root/
```
The GNU chown will exit immediately, while this chown will follow the symlink and start trying to change root.

### PR DESCRIPTION
**Pull request by water-ghosts**
Resolves #1709. Chown will now check for recursive operations on `/`, even if these occur as part of a separate operation on a directory or symlink.

This should significantly speed up the GNU test suite. Right now, the test `chown/preserve-root` takes about 7 minutes to complete, because UUtils attempts to update the owner of every single file. With this PR, the operation is aborted in less than 1 second.

### CODE REVIEW

File: tests/by-util/test_chgrp.rs [hunk 1]
```diff
@@ -104,7 +104,7 @@ fn test_preserve_root_symlink() {
             .arg("-HR")
             .arg("bin").arg(file)
             .fails()
-            .stderr_is("chgrp: it is dangerous to operate recursively on '/'\nchgrp: use --no-preserve-root to override this failsafe");
+            .stderr_contains("chgrp: it is dangerous to operate recursively");
```
Comment (sylvestre) [inline tests/by-util/test_chgrp.rs:107]: could you please run rustfmt? thanks

File: tests/by-util/test_chgrp.rs [hunk 2]
```diff
@@ -83,9 +83,10 @@ fn test_preserve_root() {
         new_ucmd!()
             .arg("--preserve-root")
             .arg("-R")
-            .arg("bin").arg(d)
+            .arg("bin")
+            .arg(d)
             .fails()
-            .stderr_is("chgrp: it is dangerous to operate recursively on '/'\nchgrp: use --no-preserve-root to override this failsafe");
+            .stderr_contains("chgrp: it is dangerous to operate recursively");
```
Comment (sylvestre) [inline tests/by-util/test_chgrp.rs:89]: why did you remove the full error message? thanks

Comment (sylvestre) [reply tests/by-util/test_chgrp.rs:89]: ping?

Comment (water-ghosts) [reply tests/by-util/test_chgrp.rs:89]: Sorry for the delay in responding! In GNU, `chgrp --preserve-root -R bin /` prints the error "chgrp: it is dangerous to operate recursively on '/'"

If the path is indirect, GNU adds more context. `chgrp --preserve-root -R bin /tmp/..` prints "chgrp: it is dangerous to operate recursively on '/tmp/..' (same as '/')"

I added the parenthetical for indirect paths that resolve to `/`, for consistency with GNU. Since different test paths now give different error messages, checking for an exact match against a static string no longer works.

**Comment by tertsdiepraam** [comment]
Does this fix https://github.com/uutils/coreutils/issues/1004?

**Comment by sylvestre** [comment]
No action for a while, closing. Please reopen when ready!
